[Permissions] Deny command with Class match

# Command bugs

#### Command name

permissions removeserverrule

#### What cog is this command from?

Permissions

#### What were you expecting to happen?

For permissions to be able to determine that I was asking for a command instead of the cog.

#### What actually happened?

For this example, I have a cog called test. The class is Test and it has a command named test. The role that I am allowing/denying to use the command is called Rolename. 

I used:

[p]permissions setdefaultglobalrule deny test
[p]permissions addserverrule allow test Rolename
[p]permissions removeserverrule test Rolename

I received: 
```
[2019-09-25 15:38:59] [ERROR] red: Exception in command 'permissions removeserverrule'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/discord/ext/commands/core.py", line 79, in wrapped
    ret = await coro(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/redbot/cogs/permissions/permissions.py", line 383, in permissions_removeguildrule
    cog_or_cmd=cog_or_command, model_id=w.id, guild_id=ctx.guild.id
  File "/usr/local/lib/python3.7/site-packages/redbot/cogs/permissions/permissions.py", line 511, in _remove_rule
    cog_or_cmd.obj.clear_rule_for(model_id, guild_id=guild_id)
  File "/usr/local/lib/python3.7/site-packages/redbot/core/commands/commands.py", line 403, in clear_rule_for
    should_continue = parent.reevaluate_rules_for(model_id, guild_id=guild_id)[1]
  File "/usr/local/lib/python3.7/site-packages/redbot/core/commands/commands.py", line 521, in reevaluate_rules_for
    for cmd in self.all_commands.values()
AttributeError: 'Test' object has no attribute 'all_commands'
```

#### How can we reproduce this issue?

Follow the instructions outlined above in the previous section.

--- a/changelog.d/3014.bugfix.rst
+++ b/changelog.d/3014.bugfix.rst
@@ -1,0 +1,1 @@
+Fixes an issue with clearing rules in permissions

--- a/redbot/core/commands/commands.py
+++ b/redbot/core/commands/commands.py
@@ -574,6 +574,14 @@ class CogMixin(CogGroupMixin, CogCommandMixin):
     """Mixin class for a cog, intended for use with discord.py's cog class"""
 
     @property
+    def all_commands(self) -> Dict[str, Command]:
+        """
+        This does not have identical behavior to 
+        Group.all_commands but should return what you expect
+        """
+        return {cmd.name: cmd for cmd in self.__cog_commands__}
+
+    @property
     def help(self):
         doc = self.__doc__
         translator = getattr(self, "__translator__", lambda s: s)

